File: Trabajo_9/JAVA/Ejercicios/Ejercicio_2.java
package Trabajo_9.Ejercicios;

interface Animal {
    void comer();
    void dormir();
    void moverse();
}

class Mamifero implements Animal {
    @Override
    public void comer() {
        System.out.println("Comiendo como un mamífero");
    }

    @Override
    public void dormir() {
        System.out.println("Durmiendo como un mamífero");
    }

    @Override
    public void moverse() {
        System.out.println("Moviendo como un mamífero");
    }
}

class Ave implements Animal {
    @Override
    public void comer() {
        System.out.println("Comiendo como un ave");
    }

    @Override
    public void dormir() {
        System.out.println("Durmiendo como un ave");
    }

    @Override
    public void moverse() {
        System.out.println("Moviendo como un ave");
    }
}

class Reptil implements Animal {
    @Override
    public void comer() {
        System.out.println("Comiendo como un reptil");
    }

    @Override
    public void dormir() {
        System.out.println("Durmiendo como un reptil");
    }

    @Override
    public void moverse() {
        System.out.println("Moviendo como un reptil");
    }
}

class Pez implements Animal {
    @Override
    public void comer() {
        System.out.println("Comiendo como un pez");
    }

    @Override
    public void dormir() {
        System.out.println("Durmiendo como un pez");
    }

    @Override
    public void moverse() {
        System.out.println("Moviendo como un pez");
    }
}

public class Ejercicio_2 {

    static void infoAnimal(Animal animal) {
        animal.comer();
        animal.dormir();
        animal.moverse();
    }
    public static void main(String[] args) {
        Animal[] animales = new Animal[4];

        animales[0] = new Mamifero();
        animales[1] = new Ave();
        animales[2] = new Reptil();
        animales[3] = new Pez();

        for (Animal animal : animales) {
            infoAnimal(animal);
        }
    }
}

File: Trabajo_9/PY/Ejercicios/Ejercicio_2.py
from abc import ABC, abstractmethod

class Animal(ABC):
    @abstractmethod
    def comer(self):
        pass
    @abstractmethod
    def dormir(self):
        pass
    @abstractmethod
    def moverse(self):
        pass

class Mamifero(Animal):
    def comer(self):
        print("El mamifero esta comiendo")

    def dormir(self):
        print("El mamifero esta durmiendo")

    def moverse(self):
        print("El mamifero esta caminando")

class Ave(Animal):
    def comer(self):
        print("El ave esta comiendo")

    def dormir(self):
        print("El ave esta durmiendo")


    def moverse(self):
        print("El ave est volando")

class Reptil(Animal):
    def comer(self):
        print("El reptil esta comiendo")

    def dormir(self):
        print("El reptil est@ durmiendo")

    def moverse(self):
        print("El reptil esta reptando.")

class Anfibio(Animal):
    def comer(self):
        print("El anfibio esta comiendo")

    def dormir(self):
        print("El anfibio esta durmiendo")

    def moverse(self):
        print("El anfibio esta saltando...")


if __name__ == "__main__":
    animales = []
    animales.append(Anfibio())
    animales.append(Reptil())
    animales.append(Ave())
    animales.append(Mamifero())

    for animal in animales:
        animal.comer()
        animal.dormir()
        animal.moverse()
    